Update dependencies

btecu opened the pull request:
Once released will allow `ember-cli` to be installed without warning:

`warning ember-cli > ember-cli-preprocess-registry > broccoli-funnel > exists-sync@0.0.4: Please replace with usage of fs.existsSync`

twokul commented on package.json:
```diff
@@ -22,16 +22,16 @@
+    "broccoli-funnel": "^2.0.1",
+    "broccoli-merge-trees": "^3.0.1",
+    "debug": "^3.0.1",
+    "ember-cli-lodash-subset": "^2.0.1",
```
do we need `ember-cli-lodash-subset` at all?

btecu replied:
Yes, it's used in two other places (`find` and `uniq`)

NullVoxPopuli replied:
find is a native function now, isn't it? (at least on arrays?)

btecu replied:
Yes, I was thinking about replacing it, although `uniq` is still needed.

btecu commented:
@twokul I've rewritten the two other use cases of `ember-cli-lodash-subset` and removed the dependency. Also remove other dependencies that were not used.

Turbo87 commented on package.json:
```diff
@@ -22,16 +22,13 @@
-    "ember-cli-lodash-subset": "^1.0.7",
-    "process-relative-require": "^1.0.0",
-    "silent-error": "^1.0.0"
+    "broccoli-clean-css": "^2.0.1",
```
it turns out that this was even more of a breaking change than I thought (https://github.com/jakubpawlowicz/clean-css#important-40-breaking-changes). we are exposing the `clean-css` options via `minifyCSS.options` and since the options have changed this in itself was a breaking change.
